src/ui/replay_parser.rs

```diff
@@ -2366,6 +2367,192 @@ impl ToolkitTabViewer<'_> {
+                "distance_traveled": report.distance_traveled
+            });
+
+            // Build combat achievements section
+            let combat = serde_json::json!({
+                "fires_caused": report.fires,
+                "floods_caused": report.floods,
+                "citadels": report.citadels,
+                "critical_hits": report.crits,
+                "spotting_damage": report.spotting_damage
+            });
+
+            // Build stats object based on enabled columns
+            let mut stats = serde_json::Map::new();
+
+            // Always include base stats
+            stats.insert("damage_total".to_string(), json!(damage_dealt["damage_total"]));
+            stats.insert("base_xp".to_string(), json!(experience["base_xp"]));
+            stats.insert("time_lived_secs".to_string(), json!(battle_life["time_lived_secs"]));
+            stats.insert("spotting_damage".to_string(), json!(combat["spotting_damage"]));
+
+            // Include damage breakdown
+            if let Some(v) = damage_dealt["damage_ap"].as_number() { stats.insert("damage_ap".to_string(), json!(v)); }
+            if let Some(v) = damage_dealt["damage_he"].as_number() { stats.insert("damage_he".to_string(), json!(v)); }
+            if let Some(v) = damage_dealt["damage_he_secondary"].as_number() { stats.insert("damage_he_secondary".to_string(), json!(v)); }
+            if let Some(v) = damage_dealt["damage_sap"].as_number() { stats.insert("damage_sap".to_string(), json!(v)); }
+            if let Some(v) = damage_dealt["damage_sap_secondary"].as_number() { stats.insert("damage_sap_secondary".to_string(), json!(v)); }
+            if let Some(v) = damage_dealt["damage_torpedoes"].as_number() { stats.insert("damage_torpedoes".to_string(), json!(v)); }
+            if let Some(v) = damage_dealt["damage_deep_water_torpedoes"].as_number() { stats.insert("damage_deep_water_torpedoes".to_string(), json!(v)); }
+            if let Some(v) = damage_dealt["damage_fire"].as_number() { stats.insert("damage_fire".to_string(), json!(v)); }
+            if let Some(v) = damage_dealt["damage_flooding"].as_number() { stats.insert("damage_flooding".to_string(), json!(v)); }
+
+            // Include potential damage
+            stats.insert("potential_damage_total".to_string(), json!(potential_damage["potential_damage_total"]));
+            if let Some(v) = potential_damage["potential_damage_artillery"].as_number() { stats.insert("potential_damage_artillery".to_string(), json!(v)); }
+            if let Some(v) = potential_damage["potential_damage_torpedoes"].as_number() { stats.insert("potential_damage_torpedoes".to_string(), json!(v)); }
+            if let Some(v) = potential_damage["potential_damage_planes"].as_number() { stats.insert("potential_damage_planes".to_string(), json!(v)); }
+
+            // Include received damage if enabled
+            if ui_report.columns.contains(&ReplayColumn::ReceivedDamage) {
```

Comment:
Why only add this if the column is enabled? Surely whatever tool is ingesting the data can decide what it wants to filter?